components/filesystem.rst

```diff
@@ -68,6 +68,10 @@ On POSIX filesystems, directories are created with a default mode value
 
     This function ignores already existing directories.
 
+.. note::
+
+    The directory permissions are affected by the current `umask<https://en.wikipedia.org/wiki/Umask/>`_.. Set the umask for your webserver, use PHP's `umask() function<https://secure.php.net/umask>`_. or use the chmod() function after the directory has been created.
```

Comment:
There are few things to change here, but overall this looks good. Can you update the paragraph to something like this:

```rst
.. note::

    The directory permissions are affected by the current `umask`_. Set
    the umask for your webserver, use PHP's :phpfunction:`umask` function
    or use the :phpfunction:`chmod` function after the directory has been
    created.
```

You then need to add the link target at the end of the document:

```rst
.. _`umask`: https://en.wikipedia.org/wiki/Umask
```